tests/unit/browser/test_adblock.py

```diff
@@ -0,0 +1,434 @@
+
+    """Mock browser.downloads.DownloadManager."""
+
+    def get(self, url, fileobj, **kwargs):
+
+        """Return a FakeDownloadItem instance with a fileobj
+           whose content is copied from the file the given url links to.
+        """
+
+        download_item = FakeDownloadItem(fileobj, name=url.path())
+        with open(url.path(), 'rb') as fake_url_file:
+            # Ensure cursors are at position 0 before copying
+            fake_url_file.seek(0)
+            download_item.fileobj.seek(0)
+            shutil.copyfileobj(fake_url_file, download_item.fileobj)
+        return download_item
+
+
+@pytest.yield_fixture
+def download_stub(win_registry):
+
+    """Register a FakeDownloadManager."""
+
+    stub = FakeDownloadManager()
+    objreg.register('download-manager', stub,
+                    scope='window', window='last-focused')
+    yield stub
+    objreg.delete('download-manager', scope='window', window='last-focused')
+
+
+def create_zipfile(directory, files, zipname='test'):
+
+    """Return a path to a created zip file
+
+       Args :
+       - directory : path object where to create the zip file
+       - files : list of paths to each file to add in the zipfile
+       - zipname : name to give to the zip file.
+    """
+
+    directory = str(directory)
+    zipfile_path = os.path.join(directory, zipname + '.zip')
+    with zipfile.ZipFile(zipfile_path, 'w') as new_zipfile:
+        for file_name in files:
+            new_zipfile.write(file_name, arcname=os.path.basename(file_name))
+            # Removes path from file name
+    return zipfile_path
+
+
+def create_blocklist(directory,
+                     blocked_hosts=BLOCKLIST_HOSTS,
+                     name='hosts',
+                     line_format='one_per_line'):
+
+    """Return a QUrl instance linking to a blocklist file.
+
+       Args:
+         - directory : path object where to create the blocklist file
+         - blocked_hosts : an iterable of string hosts to add to the blocklist
+         - name : name to give to the blocklist file
+         - line_format : 'etc_hosts'  -->  /etc/hosts format
+                         'one_per_line'  -->  one host per line format
+                         'not_correct'  -->  Not a correct hosts file format.
+    """
+
+    blocklist = QUrl(os.path.join(str(directory), name))
+    with open(blocklist.path(), 'w', encoding='UTF-8') as hosts:
+        # ensure comments are ignored when processing blocklist
+        hosts.write('# Blocked Hosts List #\n\n')
+        if line_format == 'etc_hosts': # /etc/hosts like format
+            for path in blocked_hosts:
+                hosts.write('127.0.0.1  ' + path + '\n')
+        elif line_format == 'one_per_line':
+            for path in blocked_hosts:
+                hosts.write(path + '\n')
+        elif line_format == 'not_correct':
+            for path in blocked_hosts:
+                hosts.write(path + ' This is not a correct hosts file\n')
```

Comment:
please add an `else:` here which raises a `ValueError` or so if `line_format ` is not correct.